misc/docs-linter/main.go

```diff
@@ -0,0 +1,126 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"github.com/gnolang/gno/tm2/pkg/commands"
+	"golang.org/x/sync/errgroup"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+)
+
+var (
+	ErrEmptyPath = errors.New("you need to pass in a path to scan")
+)
+
+type cfg struct {
+	docsPath string
+}
+
+func main() {
+	cfg := &cfg{}
+
+	cmd := commands.NewCommand(
+		commands.Metadata{
+			Name:       "docs-linter",
+			ShortUsage: "docs-linter [flags]",
+			ShortHelp:  "Finds broken 404 links in the .md files in the given folder & subfolders",
+		},
+		cfg,
+		func(ctx context.Context, args []string) error {
+			return execLint(cfg, ctx)
+		})
+
+	cmd.Execute(context.Background(), os.Args[1:])
+}
+
+func (c *cfg) RegisterFlags(fs *flag.FlagSet) {
+	fs.StringVar(
+		&c.docsPath,
+		"path",
+		"./",
+		"path to dir to walk for .md files",
+	)
+}
+
+func execLint(cfg *cfg, ctx context.Context) error {
+	if cfg.docsPath == "" {
+		return ErrEmptyPath
+	}
+
+	abs, err := filepath.Abs(cfg.docsPath)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("Linting %s...\n", abs)
```

Comment:
Why not use `commands.IO`?

Reply:
It seems to me that `fmt` is quite more simple and more known, and is a standard library, which is why I opted for it.